src/memory-analyzer/analyze_symbol.cpp

```diff
@@ -0,0 +1,375 @@
+/*******************************************************************\
+
+Module: Symbol Analyzer
+
+Author: Malte Mues <mail.mues@gmail.com>
+        Daniel Poetzl
+
+\*******************************************************************/
+
+#include "analyze_symbol.h"
+
+#include <util/c_types.h>
+#include <util/c_types_util.h>
+#include <util/config.h>
+#include <util/expr_initializer.h>
+#include <util/string_constant.h>
+
+gdb_value_extractort::gdb_value_extractort(
+  const symbol_tablet &symbol_table,
+  const char *binary)
+  : gdb_api(binary),
+    symbol_table(symbol_table),
+    ns(symbol_table),
+    c_converter(ns, expr2c_configurationt::clean_configuration),
+    allocate_objects(ID_C, source_locationt(), "", this->symbol_table)
```

Comment:
Use `irep_idt{}` instead of `""`, and I don't this the use of `this->` is necessary here.

Reply:
Done for the first part, but no for the second: if we don't explicitly refer to the member `symbol_table` we bind the constant reference the constructor got as its argument. And `allocate_objects` takes a non-const reference.